### ISSUE TITLE
Window size messed up after adding/removing splits

### ISSUE THREAD

**Issue by slackhead** [irssi/irssi#1031]
Irssi 1.2.0
After making and removing some split windows, and using awl script, my main window size is wrong.  I am permanently stuck in a half height window.  config shows:

```
mainwindows = {
  1 = {
    first_line = "15";
    lines = "14";
    first_column = "0";
    columns = "114";
  };
};
```

The lines ought to be 28 and first_line 0, but I can't seem to correct this.  I am also using screen.  I tried editing/removing that section, then /reload but it didn't help.

I also tried sending a sigwinch and using stty to change $LINES.

/script exec print(Irssi::windows->{'height'}) also shows the height but I can't seem to change it that way either.

**Comment by ailin-nemui** [comment]
screenshot?

**Comment by slackhead** [comment]
Sorry I had to restart irssi and I didn't think about getting a screenshot :/

All I can say is that the top half of the screen was just blank, but awl still showed at the top when I set awl_hide_data.  I was getting an error when I tried to use grow/size that there wasn't enough room to grow.

I've stopped using awl now, because I've had problems with it redrawing the screen at times when using split windows and the hide data settings, but this is the first time I've ever had the window size stick like that.  I'm using a custom statusbar at the top with 'act' in it.

**Comment by ailin-nemui** [comment]
were you using adv_windowlist in statusbar mode? If you had some recipe that could reproduce this issue it would possibly be easier to tackle. Do you have a history of the commands you typed? Did you try physically resizing the terminal window?

**Comment by slackhead** [comment]
Yes, it was in statusbar mode.  I also had nicklist.pl running.  The redrawing problem happened only two or three times after irssi had been running a few days.  The screen size problem only happened the once.  A quick grep of awl in config:

```
awl_shared_sbar = "OFF";
awl_viewer = "no";
awl_placement = "top";
awl_sbar_maxlength = "no";
awl_hide_empty = "0";
awl_hide_data = "2";
awl_height_adjust = "2";
awl_all_disable = "yes";
awl_detach = " ";
awl_detach_aht = "no";
awl_no_mode_hint = "no";
awl_viewer_launch = "no";
awl_custom_xform = "";
awl_maxcolumns = "0";
awl_block = "0";
awl_maxlines = "0";
awl0 = "/^set awl_hide_data 0; /^nicklist update";
awl2 = "/^set awl_hide_data 2; /^nicklist update";
awl3 = "/^set awl_hide_data 3; /^nicklist update";
{ key = "^[OP"; id = "command"; data = "^set awl_hide_data 0"; },
{ key = "^[OQ"; id = "command"; data = "^set awl_hide_data 2"; },
{ key = "^[OR"; id = "command"; data = "^set awl_hide_data 3"; },
```

I did try adjusting some settings like maxlines and block to see if I could fix the issue.  I would mostly have it running with awl_hide_data = 2, but switch to 0 at times to see the full window list.

I would normally only have one split made by /window show and removed by an alias: hide = /^window stick $0 off; /^window hide $0

I have binds for /window grow and /window shrink.

That's everything significant that I can think of.

**Comment by ailin-nemui** [comment]
it doesn't seem likely that this is related to awl;

did the issue happen after a /win grow/shrink or did it happen "by itself"?

**Comment by slackhead** [comment]
I'm a frequent grower/shrinker so it was probably after one of those.

**Comment by ailin-nemui** [comment]
If you see it again try to gather some more evidence before restarting :-) for clarification this was with 1.2.0 release right?

**Comment by slackhead** [comment]
Yes 1.2.0 and I'll get more evidence next time.
